### PR TITLE
Clarify docs for version constraints

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java [hunk 1]
```diff
@@ -44,12 +44,27 @@
      */
     void setBranch(@Nullable String branch);
 
+    /**
+     * Sets the version as strict.
+     * <p>
+     * Any version not matched by this version notation will be excluded. This is the strongest version declaration.
+     * It will cause dependency resolution to fail if no version acceptable by this clause can be selected.
+     * This term supports dynamic versions.
+     * This clears any set rejected versions.
```
Comment (ljacomet) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java:53]: I would move this sentence below the next one and make it a different paragraph as well.

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java [hunk 2]
```diff
@@ -44,12 +44,27 @@
...
+    void strictly(String version);
+
     /**
      * Sets the required version of this module.
      * <p>
      * Implies that the selected version cannot be lower than what {@code require} accepts but could be higher through conflict resolution, even if higher has an exclusive higher bound.
-     * This is what a direct version on a dependency translates to.
+     * This is what a direct dependency translates to.
      * This term supports dynamic versions.
+     * This clears any set rejected versions.
```
Comment (ljacomet) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java:67]: I would move this sentence below the next one and make it a different paragraph as well.

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java [hunk 3]
```diff
@@ -64,29 +79,17 @@
      * This is a very soft version declaration.
      * It applies only if there is no stronger non dynamic opinion on a version for the module.
      * This term does not support dynamic versions.
+     * This clears any set rejected versions.
```
Comment (ljacomet) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java:82]: I would move this sentence below the next one and make it a different paragraph as well.

**Review by ljacomet — COMMENTED** [review]
Minor suggestions for the Javadoc part:

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**